### PR TITLE
[HIVEMALL-65] Update define-all.spark and import-packages.spark

### PR DESCRIPTION
**Pull request by wangyum**
## What changes were proposed in this pull request?

Make define-all.spark keep correspondence with define-all.hive.

## What type of PR is it?

Bug Fix

## What is the Jira issue?

https://issues.apache.org/jira/browse/HIVEMALL-65

## How was this patch tested?

manual tests

### CODE REVIEW

**Comment by maropu** [comment]
@amaya382 Could you review this?

**Comment by amaya382** [comment]
@maropu Sure

**Comment by wangyum** [comment]
I'm not sure whether these functions should add to `define-all.spark`:
https://github.com/apache/incubator-hivemall/blob/master/resources/ddl/define-all.hive#L467-L483

**Comment by maropu** [comment]
Aha, can you replace `RowIdUDF` with `RowIdUDFWrapper`?: https://github.com/apache/incubator-hivemall/blob/master/spark/spark-common/src/main/java/hivemall/tools/mapred/RowIdUDFWrapper.java
And then, you can remove all the entries there except for `rowid`.

**Comment by wangyum** [comment]
OK, 
`hivemall.smile.regression.RandomForestRegressionUDTF` create 2 functions. `define-all.spark` also add them?
https://github.com/apache/incubator-hivemall/blob/master/resources/ddl/define-all.hive#L607-L611

**Comment by maropu** [comment]
yea, please.

**Comment by myui** [comment]
@maropu you can revise taskid heneration scheme. http://stackoverflow.com/questions/31631791/how-to-get-id-of-a-map-task-in-spark

**Comment by wangyum** [comment]
OK, all done

**Comment by maropu** [comment]
@myui No, you can't access `TaskContext` inside `HiveUDF`.

**Comment by myui** [comment]
@maropu You can use Reflections.

**Comment by maropu** [comment]
Aha, I see. I'll make a JIRA

**Comment by maropu** [comment]
Created: https://issues.apache.org/jira/browse/HIVEMALL-68

File: resources/ddl/define-all.spark [hunk 1]
```diff
@@ -83,15 +101,15 @@ sqlContext.sql("CREATE TEMPORARY FUNCTION distance2similarity AS 'hivemall.knn.s
...
-sqlContext.sql("DROP TEMPORARY FUNCTION IF EXISTS homming_distance")
-sqlContext.sql("CREATE TEMPORARY FUNCTION hamming_distance AS 'hivemall.knn.distance.HammingDistanceUDF'")
-
 sqlContext.sql("DROP TEMPORARY FUNCTION IF EXISTS popcnt")
 sqlContext.sql("CREATE TEMPORARY FUNCTION popcnt AS 'hivemall.knn.distance.PopcountUDF'")
 
 sqlContext.sql("DROP TEMPORARY FUNCTION IF EXISTS kld")
 sqlContext.sql("CREATE TEMPORARY FUNCTION kld AS 'hivemall.knn.distance.KLDivergenceUDF'")
 
+sqlContext.sql("DROP TEMPORARY FUNCTION IF EXISTS homming_distance")
```
Comment (myui) [inline resources/ddl/define-all.spark]: @wangyum Typo. `hamming_distance` is correct.

Comment (wangyum) [inline resources/ddl/define-all.spark]: fixed it.

**Comment by wangyum** [comment]
@maropu , @myui I have sent an email to `dev@hivemall.incubator.apache.org`. have you received it ?

**Comment by myui** [comment]
LGTM. 

@maropu merge this one if @amaya382 says fine.

**Comment by maropu** [comment]
okay

**Comment by amaya382** [comment]
LGTM
Notice: We also need revise `import-packages.spark` in [HIVEMALL-65]. @wangyum Can you do it? (or postpone to other PR)

**Comment by myui** [comment]
@amaya382 what kind of imports are missing in `import-packages.spark` ? (cc: @maropu )

**Comment by amaya382** [comment]
@myui it seems some imports (`HmLabeledPoint` and `HmFeature`) should be removed and some ones should be revised to correct package name, but i haven't check it well yet... maropu should know in detail

**Comment by myui** [comment]
@maropu Could you merge this PR? `import-packages.spark` should be another ticket.

**Comment by wangyum** [comment]
`import-packages.spark` already updated:
https://github.com/apache/incubator-hivemall/pull/44/commits/1a48d980d4ebcaf8df6d861edc27ff80dfd3d261

**Comment by maropu** [comment]
LGTM. I'll merged later

**Comment by myui** [comment]
@wangyum Merged. Thank you for your contribution!

**Pull request closed without merging**